### ISSUE TITLE
[BUG] When publishing the PowerBI report, there is no way to perform or schedule refreshes

### ISSUE THREAD

**Issue by cristianoag** [microsoft/powerapps-tools#446]
When publishing the Power BI reports for BAR App, we are unable to schedule or perform refreshes. The following message is presented:

You can't schedule refresh for this dataset because the following data sources currently don't support refresh:
Data source for Query1

**Comment by bungleweed** [comment]
Yes same issue here. Have you found a workaround at all? Also looks like a duplicate of this issue https://github.com/microsoft/powerapps-tools/issues/390?

**Comment by gjmscc** [comment]
All the queries that use http://www.webcal.fi will cause the error:
Requested Date
Check-in Date
Check-out Date 

Reasons for error: https://blog.crossjoin.co.uk/2016/08/23/web-contents-m-functions-and-dataset-refresh-errors-in-power-bi/

Fixed this using this syntax:  https://blog.crossjoin.co.uk/2016/08/16/using-the-relativepath-and-query-options-with-web-contents-in-power-query-and-power-bi-m-code/

  Example M Code for the web.content queries that works 

```
    PHSource = Xml.Tables(Web.Contents(
                              "http://www.webcal.fi",
                               [
                                    RelativePath="/cal.php",
                                    Query=
                                    [
                                        id="408",
                                        format="xml",
                                        start_year=Text.From(StartYear),
                                        end_year=Text.From(EndYear),
                                        tz="Europe%2FLondon"
                                    ] 
                                ]
                    )),

```

**Comment by katrinar50** [comment]
Has anyone had any joy with applying the fix above? We have tried it but now whilst it's fine in Desktop,and when first published to the Service the report shows today's data. However, with the fix in place when a manual or scheduled refresh of the dataset is completed it will then show tomorrows data in the report.   The Service and report all use UTC time so it shouldn't be that but we have changed time zone in the fix, tried it without but no difference. The issue is shown in the document below.
[Building Access BI data refresh error.docx](https://github.com/microsoft/powerapps-tools/files/5428658/Building.Access.BI.data.refresh.error.docx)

**Comment by gjmscc** [comment]
The year,  month and date value in your example which query is that value coming from ? Is **Requests** and the **RequestDate** column ?

Have you found that since the timezone change in the UK over the weekend the dates are correctly aligned between the PBI Service and PBI Desktop?

**Comment by dabaggett** [comment]
Are you going to provide the code to move from Webcal.fi to Webcal.guru? 

Thank you for the code snippet above.  Worked perfectly allowing the calendar to be scheduled online.  It notes those specific queries will not refresh but allows the others to.  Since those queries obtain holidays they would not need to be run from the desktop very often since it has 5 years of holidays.
